ballerina/swiftmt_parser.bal

```diff
@@ -0,0 +1,272 @@
+        } else {
+            transactionXml += tagElement;
+        }
+    }
+    xmlArray.push(transactionXml);
+    xml newBlock4Xml = xml ``;
+
+    if messageType.equalsIgnoreCaseAscii("201") || messageType.equalsIgnoreCaseAscii("203") {
+        newBlock4Xml = xmlArray[0];
+    } else {
+        newBlock4Xml = xmlArray[1];
+    }
+
+    if (messageType + validationFlag).equalsIgnoreCaseAscii("202COV") || (messageType + validationFlag).equalsIgnoreCaseAscii("205COV") {
+        xml:Element rootXml = xml `<UndrlygCstmrCdtTrf/>`;
+        rootXml.setChildren(xmlArray[2]);
+        newBlock4Xml += rootXml;
+    }
+
+    foreach int index in 1 ... xmlArray.length() - 1 {
+        if messageType.equalsIgnoreCaseAscii("201") || messageType.equalsIgnoreCaseAscii("203") || (messageType.equalsIgnoreCaseAscii("204") && index > 1) {
+            xml:Element rootXml = xml `<Transaction/>`;
+            rootXml.setChildren(xmlArray[index]);
+            newBlock4Xml += rootXml;
+        }
+    }
+
+    foreach xml block in customXml.elementChildren() {
+        if block.getName().equalsIgnoreCaseAscii("block4") {
+            block.setChildren(newBlock4Xml);
+        }
+    }
+    return customXml;
+}
+
+# This function adds a copy of the original message content into the block 4 of a SWIFT message.
+# The original message is identified after encountering the "21" tag, and certain tags (e.g., "11S", "11R", "75", "76", "77A", "79") are excluded from the copy.
+# The copied original message is then encapsulated in a new XML tag `<MessageCopy>` and appended to block 4.
+#
+# + customXml - The input XML representing the SWIFT message with multiple tags.
+# + return - Returns the modified XML with the added `<MessageCopy>` tag, which contains a copy of the original message content.
+isolated function addCopyOfOriginalMessage(xml customXml) returns xml {
+    xml newBlock4Xml = xml ``;
+    xml originalMessageXml = xml ``;
+    boolean isOriginalMessage = false;
+
+    foreach xml tagElement in (customXml/**/<block4>).elementChildren() {
+        string name = tagElement.elementChildren("name").data();
+        if name.equalsIgnoreCaseAscii("21") && !isOriginalMessage {
+            newBlock4Xml += tagElement;
+            isOriginalMessage = true;
+        } else if (!(name.equalsIgnoreCaseAscii("11S")) || !(name.equalsIgnoreCaseAscii("11R")) || !(name.equalsIgnoreCaseAscii("75")) || !(name.equalsIgnoreCaseAscii("76")) || !(name.equalsIgnoreCaseAscii("77A")) || !(name.equalsIgnoreCaseAscii("79"))) && isOriginalMessage {
+            originalMessageXml += tagElement;
+        } else {
+            newBlock4Xml += tagElement;
+        }
+    }
+    newBlock4Xml += xml `<MessageCopy>${originalMessageXml}</MessageCopy>`;
+
+    foreach xml block in customXml.elementChildren() {
+        if block.getName().equalsIgnoreCaseAscii("block4") {
+            block.setChildren(newBlock4Xml);
+        }
+    }
+    return customXml;
+}
```

Comment:
As a general comment: I am more prefer to declare this string values as constants and using it

Reply:
But this string content is getting changed.